Unity scripts/WebcamPublisher.cs

```diff
@@ -0,0 +1,102 @@
+/*
+� CentraleSupelec, 2017
+Author: Dr. Jeremy Fix (jeremy.fix@centralesupelec.fr)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+<http://www.apache.org/licenses/LICENSE-2.0>.
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Adjustments to new Publication Timing and Execution Framework 
+// � Siemens AG, 2018, Dr. Martin Bischoff (martin.bischoff@siemens.com)
+
+using UnityEngine;
+
+namespace RosSharp.RosBridgeClient
+{
+    public class WebcamPublisher : UnityPublisher<MessageTypes.Sensor.CompressedImage>
+    {
+        //public Camera ImageCamera;
+        public string FrameId = "Camera";
+        public int resolutionWidth = 640;
+        public int resolutionHeight = 480;
+        [Range(0, 100)]
+        public int qualityLevel = 50;
+
+        private MessageTypes.Sensor.CompressedImage message;
+        //private Texture2D texture2D;
+        //private Rect rect;
+
+        WebCamTexture webcam;
+        Texture2D webcamImage;
+
+        protected override void Start()
+        {
+            base.Start();
+
+            WebCamDevice[] devices = WebCamTexture.devices;
+            for (int i = 0; i < devices.Length; i++)
+                Debug.Log(devices[i].name);
+            //start webcam feed
+            webcam = new WebCamTexture(devices[0].name);
+            webcam.Play();
+            Debug.LogFormat("webcam: {0} {1} x {2}", webcam.deviceName, webcam.width, webcam.height);
+            //data = new Color32[webcam.width * webcam.height];
+            webcamImage = new Texture2D(webcam.width, webcam.height);
+
+            //InitializeGameObject();
+            InitializeMessage();
+            //Camera.onPostRender += UpdateImage;
+            UpdateMessage();
+        }
+
+        private void Update()
+        {
+            UpdateMessage();
+        }
+
+        /*private void UpdateImage(Camera _camera)
```

Comment:
These are sections from the original Siemens code that were commented out -- is it educative to retain them, or should the file be clean without commented lines of code?